docs/examples/sujoy_edge/sujoy_edge_demo.jl

```diff
@@ -0,0 +1,80 @@
+# ---
+# cover: assets/sedge.png
+# title: Sujoy Edge Demo
+# ---
+
+using Images,Statistics,ImageView,TestImages
+
+# Sujoy edge function
+
+
+# [Paper Link] (https://www.ijert.org/research/a-better-first-derivative-approach-for-edge-detection-IJERTV2IS110616.pdf)
+
+## edges = sujoy(img, kernel_nhood=0, just_filter=False),  
+## 'edges' is a image of edges computed according to Sujoy's algorithm,  
+
+
+## Parameters
+## ----------
+## img : any gray image,
+## kernel_nhood : 0(default) or 1,
+## if 0, kernel is based on 4-neighborhood,
+## else, kernel is based on 8-neighborhood,
+## just_filter : boolean(optional),
+## if true, then return the result of filtering the image with
+## the Sujoy's filter(s), but does not threshold (default is False).
+##
+## Returns
+## -------
+## edges : gray image (if just_filter is true),
+## else, bool image.
+
+
+
+function sujoy(img, kernel_nhood=0, just_filter=false)
+    img_channel = channelview(img)
+
+    if size(img_channel)[1] == 3
+        img = Gray.(img)
+        img_channel = channelview(img)
+    end
```

Comment:
The first six lines can be simply replaced by this:

```suggestion
img_channel = Gray.(img)
```

This is better than manually check the first channel.

Reply:
Fix this